tests/resources/serializers/test_csl_serializer.py

```diff
@@ -39,3 +43,56 @@ def test_csl_serializer(
     serialized_record = serializer.dump_one(full_record)
 
     assert serialized_record == expected_data
+
+
+def test_citation_string_serializer(
+        running_app, full_record, vocabulary_clear):
+    """Test Citation String Serializer."""
+    # if the record is created this field will be present
+    full_record["id"] = "12345-abcde"
+
+    expected_data = "Nielsen, Lars Holm (2018–2020) InvenioRDM. InvenioRDM. doi: 10.5281/inveniordm.1234."  # noqa
+
+    serializer = StringCitationSerializer()
+    serialized_record = serializer.serialize_object(full_record)
+    assert serialized_record == expected_data
+
+
+def test_citation_string_serializer_request(
+        running_app, full_record, vocabulary_clear, client,
+        es_clear, minimal_record, superuser_identity):
+
+    service = current_rdm_records.records_service
+    draft = service.create(superuser_identity, minimal_record)
+    record = service.publish(draft.id, superuser_identity)
+    id = record.id
+    headers = {"Accept": "text/x-bibliography"}
+
+    default_style = StringCitationSerializer._default_style
+    default_locale = StringCitationSerializer._default_locale
+
+    _url = f"/records/{id}"
+
+    test_cases = [
```

Comment:
This is nice! But I see an issue on debugging/understanding what could fail. There would be no proper message. Taking into account that most (if not all) fixtures are module-scoped. There is no penalty in using `pytest.mark.parametrize`
e.g. https://github.com/inveniosoftware/invenio-rdm-records/blob/c321c06893b92eb087cfcb3ba0f32b187aadf6fa/tests/services/schemas/test_location.py#L43 (there are more in that tests folder).

Reply:
It was me. We can use `parametrize`, but it will clean/re-create fixtures all the times. In ILS we basically removed all of them and replaced with this pattern to increase tests speed.
The drawback is that it is not easy to debug... you will have to put print/breakpoints.

Reply:
Yeah... Most of the fixtures seem to be `module` scope tho... but it's true that many others have `pytest.fixture` which will default to `function`.... This package takes ages to test already, so I agree with you lets sacrifice a bit of debugging for some speed :)